scrapy/settings/__init__.py

```diff
@@ -88,19 +103,48 @@ def getdict(self, name, default=None):
             value = json.loads(value)
         return dict(value)
 
+    def _getcomposite(self, name):
+        # DO NOT USE THIS FUNCTION IN YOUR CUSTOM PROJECTS
+        # It's for internal use in the transition away from the _BASE settings and
+        # will be removed along with _BASE support in a future release
+        basename = name + "_BASE"
+        if basename in self:
+            warnings.warn('_BASE settings are deprecated.',
+                          category=ScrapyDeprecationWarning)
+            compsett = BaseSettings(self[name + "_BASE"], priority='default')
+            compsett.update(self[name])
+            return compsett
+        else:
+            return self[name]
+
+    def getpriority(self, name):
+        prio = None
+        if name in self:
+            prio = self.attributes[name].priority
+        return prio
+
+    def maxpriority(self):
+        if len(self) > 0:
+            return max(self.getpriority(name) for name in self)
+        else:
+            return get_settings_priority('default')
+
+    def __setitem__(self, name, value):
+        self.set(name, value)
```

Comment:
Really minor detail, but a test for this alias would be nice. Coverage of your changes is 100% otherwise, good work :)